clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -264,7 +264,7 @@ export class HttpConnection implements IConnection {
 
     private resolveUrl(url: string): string {
         // startsWith is not supported in IE
-        if (url.lastIndexOf("https://", 0) === 0 || url.lastIndexOf("http://", 0) === 0) {
+        if (url.lastIndexOf("https://", 0) === 0 || url.lastIndexOf("http://", 0) === 0 || url.lastIndexOf("//", 0) === 0) {
```

Comment:
This is going to fail on Node. It will return `//foo`, which isn't a valid URL. Node clients will get to line 271 and then throw, which makes sense, there's no "page" to resolve URLs relative to, so having Node clients throw when given a non-absolute URL (even one using `//`) is preferred.

Instead, let's refactor the part **below** the test for `window` further down, to use the `a` tag we're using, but more aggressively. I was doing some testing and I found that the following code pretty consistently normalizes URLs in Chrome (at least):

```javascript
const parser = document.createElement("a");
parser.href = url;
return parser.href; // Re-reading the property gives you the absolute URL, resolved by the browser.
```

As for testing, since we're testing browser-bound logic (URL resolution), the tests would need to be in our browser functional test suite. I'd still suggest using a test HttpClient though, since we don't really want to make real requests, just check that the URL resolution logic works.

Reply:
As an added bonus, that will also resolve https://github.com/aspnet/SignalR/issues/1835 ; two birds, one stone ;)